src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -316,31 +317,28 @@ def get_suggested_assignees(
     return []
 
 
-def get_suspect_commit_text(
-    project: Project, event: GroupEvent, commits: Sequence[Mapping[str, Any]] | None = None
-) -> SlackBlock:
+def get_suspect_commit_text(group: Group) -> str | None:
     """Build up the suspect commit text for the given event"""
 
-    # commits is passed from context when the rule initially fires
-    # we may not have that data if the message is being built after an action is taken, for example
-    if not commits:
-        commits = get_commits(project, event)
-    if not commits:
+    commit = group.get_suspect_commit()
+    if not commit:
         return None
 
-    commit = commits[0]  # get the most recent commit
     suspect_commit_text = "Suspect Commit: "
-    pull_request = commit.get("pull_request")
-    author = commit.get("author")
-    commit_id = commit.get("id")
+    pull_request = PullRequest.objects.filter(
+        merge_commit_sha=commit.key, organization_id=group.project.organization_id
+    ).first()
+
+    author = commit.author
+    commit_id = commit.key
     if not (author and commit_id):  # we need both the author and commit id to continue
         return None
 
-    author_display = author.get("name") if author.get("name") is not None else author.get("email")
+    author_display = author.name if author.name else author.email
     if pull_request:
-        repo = pull_request.get("repository", {})
-        repo_base = repo.get("url")
-        provider = repo.get("provider", {}).get("id")
+        repo = Repository.objects.get(id=pull_request.repository_id)
```

Comment:
should we do the `get_related_object(Repository)` on the original query to help save on doing another db query here?